### DISCUSSION TITLE
Is there no SignalReceived activity in Elsa 3.1.3 designer?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5762] [General]
Is there no SignalReceived activity in Elsa 3.1.3 designer?

**Comment by glime-ai** [comment]
No, in Elsa 3, the activities that were previously known as "Send Signal" and "Signal Received" in Elsa 2 have been replaced with different concepts and activities. The signaling mechanism in Elsa 3 is not the same as in Elsa 2. Instead, you might use the "Publish Event" and "Event" activities for similar purposes in Elsa 3. The "Publish Event" activity allows you to publish an event with a name and correlation ID, while the "Event" activity acts as an event handler that can continue the workflow if an event is raised.

If you are looking to implement custom logic before suspending the workflow, you may need to create custom activities. For more information on the available activities in Elsa 3 and how to use them, you can refer to the official Elsa documentation, specifically the sections on primitive activities, flow activities, and branching activities. If you need to create custom activities, you can follow the guidelines provided in the documentation and examples from the community.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by amirhaghajani** [comment]
where can I find a simple example of using Publish Event and Event activity to handle it?
